app/src/main/kotlin/view/graph/GraphView.kt

```diff
@@ -0,0 +1,64 @@
+package view.graph
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.gestures.detectTransformGestures
+import androidx.compose.foundation.gestures.panBy
+import androidx.compose.foundation.gestures.rememberTransformableState
+import androidx.compose.foundation.gestures.zoomBy
+import androidx.compose.foundation.layout.*
+import androidx.compose.material.FloatingActionButton
+import androidx.compose.material.Text
+import androidx.compose.runtime.*
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.geometry.Offset
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.graphicsLayer
+import androidx.compose.ui.input.pointer.pointerInput
+import androidx.compose.ui.platform.LocalDensity
+import androidx.compose.ui.unit.dp
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import viewmodel.graph.GraphViewModel
+
+@Composable
+fun <D> GraphView(viewModel: GraphViewModel<D>, currentScaleState: MutableState<Float>) {
+    val offset = remember { mutableStateOf(Offset.Zero) }
+    val coroutineScope = rememberCoroutineScope { Dispatchers.Default }
+
+    val updateRequired by remember { derivedStateOf { viewModel.updateIsRequired } }
```

Comment:
I use it, bc I need to trigger the recomposition in case the viewmodel's `"updateIsRequired"` state has changed. As i understood, the regular `remember  { mutableStateOf }` can't give us such functionality.

Reply:
But `viewModel.updateIsRequired` is itself a ` MutableState`. When its value changes it should trigger recomposition. I mean, if you remove `updateRequired` all together and use `viewModel.updateIsRequired` directly instead it should work just fine.

Reply:
I've tried, but it didn't work(